modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/gatewayPolicy/GatewayPolicyTestCase.java

```diff
@@ -0,0 +1,321 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.am.integration.tests.gatewayPolicy;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.conn.ssl.AllowAllHostnameVerifier;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.APIOperationPoliciesDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.GatewayPolicyDeploymentDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.GatewayPolicyMappingInfoDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.GatewayPolicyMappingsDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.OperationPolicyDTO;
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationDTO;
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationKeyDTO;
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationKeyGenerateRequestDTO;
+import org.wso2.am.integration.test.Constants;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationConstants;
+import org.wso2.am.integration.test.utils.bean.APIRequest;
+import org.wso2.am.integration.tests.api.lifecycle.APIManagerLifecycleBaseTest;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+
+import javax.xml.xpath.XPathExpressionException;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class GatewayPolicyTestCase extends APIManagerLifecycleBaseTest {
+
+    private final String API_NAME = "GatewayPolicyApplicableAPITest";
+    private final String API_CONTEXT = "gatewayPolicyApplicableAPITest";
+    private final String API_END_POINT_POSTFIX_URL = "xmlapi";
+    private String applicationId;
+    private String apiId;
+    String gatewayPolicyId;
+    String newGatewayPolicyId;
+    private String accessToken;
+    private Map<String, String> policyMap;
+
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
```

Comment:
dont we need run the test cases for tenants as well?

Reply:
This will be added via a new PR since the testing needs some time with tenant changes.